airflow/providers/google/cloud/transfers/s3_to_gcs.py

```diff
@@ -21,9 +21,13 @@
 
 from airflow.exceptions import AirflowException
 from airflow.providers.amazon.aws.hooks.s3 import S3Hook
-from airflow.providers.amazon.aws.operators.s3_list import S3ListOperator
 from airflow.providers.google.cloud.hooks.gcs import GCSHook, _parse_gcs_url, gcs_object_is_directory
 
+try:
+    from airflow.providers.amazon.aws.operators.s3 import S3ListOperator
+except ImportError:
+    from airflow.providers.amazon.aws.operators.s3_list import S3ListOperator
```

Comment:
We dont do that in other operators.
We always import from most updated path

Reply:
This will be problematic as we do not yet have the technical ability to enforce a specific provider version. We would have to add some code to add this ability. I think the current solution is sufficient and even more end-user friendly. It is also in line with how we approach backward compatibility with the airflow core.

Reply:
Actually we have the technicall capability and we already used it (min dependencies between sftp and ssh I believe) but in this case I agree breaking cross-compatibility is not a good idea.

Reply:
It is more user friendly. I just mentioned that this is the first time we do that...

On the other hand we have so many deprecated classes/functions/params right now supporting all of them is not that easy. Maybe we should consider some cleanup. Google provider for example had several major releases without removing deprecated code.